### PR TITLE
Fix config's automatic conversion of an attribute to bool gives unexpected behavior

### LINKED ISSUE #9009: config's automatic conversion of an attribute to bool gives unexpected behavior
When using the automatic conversion of a config attribute to a bool, such as with:
```
bool b = cfg["some_attribute"]
```
The variable `b` will always be false regardless of the attribute's actual value. This is due to calling `operator int()` in `config_attribute_value.hpp` instead of `explicit operator bool()`, which results in using the `attribute_numeric_visitor` which uses the default value of 0 for bool values instead of its actual value.

Potential solutions mentioned by celticminstrel from IRC:
* Remove the explicit annotation on operator bool().
* Remove operator bool() entirely, and also remove operator int().
* Remove operator bool() and replace operator int() with a custom function using a different visitor that returns 0 or 1 if the value is a bool.

Any fix for this issue should also include test cases added to the boost unit tests for this behavior as well.

### PR DESCRIPTION
**Pull request by Mahoyomu**
### Goal of the PR
- Resolves #9009.
### How does the PR work?
- By making int conversion explicit.
### Does it resolve any reported issue?
- #9009
### If not a bug fix, why is this PR needed? What use cases does it solve?
- This is a bug fix.

## To do

If there's a need for related test cases in `BOOST_UNIT_TEST`, please share with me about the details of need.

This PR is a Work In Progress.

## How to test
see changes in the src.

### CODE REVIEW

**Comment by Wedge009** [comment]
Set to draft, since this is declared a WIP.

File: src/tests/test_config.cpp [hunk 1]
```diff
@@ -404,4 +403,38 @@ BOOST_DATA_TEST_CASE(add_child_NonEmptyThis_newOrExistingKey_lOrRValue_AppendAnd
 	BOOST_CHECK_NE(&new_child, &update);
 }
 
+BOOST_AUTO_TEST_CASE(auto_conversion_bool_and_int)
+{
+	config cfg;
+	cfg["boolAttributeTrue"] = true;
```
Comment (Pentarctagon) [inline src/tests/test_config.cpp:409]: Please use snake case.

Comment (CelticMinstrel) [reply src/tests/test_config.cpp:409]: In other words:
```suggestion
	cfg["bool_attribute_true"] = true;
```
(I have no idea what the reasoning for this is, but I personally don't care one way or the other.)

Comment (Pentarctagon) [reply src/tests/test_config.cpp:409]: It's just for consistency. Nearly all of wesnoth's code is in snake case, aside from a few bits of ancient code that have stuck around.

Comment (CelticMinstrel) [reply src/tests/test_config.cpp:409]: Well, it's debatable whether a string key in a config counts as code, but sure.

**Pull request closed without merging**